2385 Added DocketAlerts API

albertisfu opened the pull request:
DocketAlerts API added as required in #2385 

- Support for Docket Alerts CRUD operations on operations `/api/rest/v3/docket-alerts/`
- Added documentation for docket alerts endpoint in REST docs and linked from Webhooks docs.

Let me know what you think.

mlissner commented on cl/api/templates/rest-docs-vlatest.html:
```diff
@@ -986,6 +986,23 @@ <h3 id="alerts-endpoint">{% url "alert-list" version="v3" %}</h3>
     <p>As always, for more details place an <code>OPTIONS</code> request to the endpoint.</p>
 
+    <h3 id="docket-alerts-endpoint">{% url "docket-alert-list" version="v3" %}</h3>
+    <p><a href="{% url "alert_help" %}#recap-alerts">Docket Alerts</a> can be made, viewed, modified, and deleted using this endpoint. This can be useful for organizations using <a href="{% url 'webhooks_docs' %}#docket-alerts">Docket Alert Webhooks</a> or if you need docket alerts on many dockets. To create a docket alert, send:
+    </p>
+    <ul>
+      <li><strong><code>docket</code></strong> &mdash; The docket to subscribe to.</li>
+      <li><strong><code>alert_type</code></strong> &mdash; Optional, by default <code>1</code> Subscription or <code>0</code> Unsubscription (docket alert disabled).</li>
+    </ul>
+    <p>For more details place an <code>OPTIONS</code> request to the endpoint.</p>
```
We need to explain what an unsubscription is in this context. How's this:

```suggestion
    <p><a href="{% url "alert_help" %}#recap-alerts">Docket Alerts</a> can be made, viewed, modified, and deleted using this endpoint. This can be useful for organizations using <a href="{% url 'webhooks_docs' %}#docket-alerts">Docket Alert Webhooks</a> or if you need docket alerts on many dockets.
    </p>
    <p>To create a docket alert, send a POST request with the <code>docket</code> ID you wish to subscribe to.
    </p>
    <p>An optional secondary argument, <code>alert_type</code>, may also be sent. This argument works with our auto-subscribe feature. When auto-subscribe is enabled in your account and your <a href="XXX">@recap.email</a> email address receives a notification from PACER, we will automatically subscribe you to that case. If instead you wish to <em>not</em> get notifications for that case, while keeping auto-subscribe on, you must "unsubscribe" from the case. To do this, create a docket alert with the <code>alert_type</code> set to <code>1</code>, which represents an unsubscription.
    </p>
    <p>For more details place an <code>OPTIONS</code> request to the endpoint.</p>
```

albertisfu replied:
Thanks, just tweaked the alert_type for unsubscription that's `0`.

mlissner reviewed (changes requested):
This is good. Two documentation notes:

1. The api docs need a link from the Table of Contents to the new endpoint.

2. I noticed that help/recap/email/ talks about webhooks, but doesn't link to their docs. Can you fix that, please?

And a few more little things in the comments. :)

albertisfu commented:
Great, I've applied the requested changes, including:

- Added the endpoint to the table content
- Linked webhook docs from recap.email help
- Added the context about unsubscription
- Changed the endpoint to use IDs instead of links

Thank you.

mlissner commented:
Great, thank you!
